### PR TITLE
Lengthen names

### PR DESCRIPTION
**Pull request by knoxsp**

### CODE REVIEW

File: hydra_base/db/model.py [hunk 1]
```diff
@@ -467,9 +468,10 @@ class TemplateType(Base, Inspect):
     )
 
     id = Column(Integer(), primary_key=True, nullable=False)
-    name = Column(String(60),  nullable=False)
+    name = Column(String(200),  nullable=False)
+    description = Column(String(1000))
```
Comment (jetuk) [inline hydra_base/db/model.py]: Is 1000 characters enough for a description?

File: hydra_base/lib/template.py [hunk 2]
```diff
@@ -116,23 +117,19 @@ def _get_attr_by_name_and_dimension(name, dimension):
...
-            dimension = units.get_unit_dimension(attribute.find('unit').text)
-
-    if dimension is None or dimension.lower() in ('dimensionless', ''):
-        dimension = 'dimensionless'
+            dimension_i = units.get_unit_dimension(attribute.find('unit').text)
 
     name      = attribute.find('name').text.strip()
 
-    attr = _get_attr_by_name_and_dimension(name, dimension)
+    attr = _get_attr_by_name_and_dimension(name, dimension_i.name)
```
Comment (jetuk) [inline hydra_base/lib/template.py]: It is possible that `dimension_i` could be unassigned here.

File: hydra_base/lib/template.py [hunk 3]
```diff
@@ -215,25 +212,17 @@ def parse_xml_typeattr(type_i, attribute):
 def parse_json_typeattr(type_i, typeattr_j, attribute_j, default_dataset_j):
 
     if attribute_j.dimension is not None:
```
Comment (jetuk) [inline hydra_base/lib/template.py]: Why not use the `parse_attribute` function? The code is repeated?

File: hydra_base/lib/template.py [hunk 4]
```diff
@@ -215,25 +212,17 @@ def parse_xml_typeattr(type_i, attribute):
...
-            dimension = units.get_unit_dimension(attribute_j.unit)
-
-    if dimension is None or dimension.lower() in ('dimensionless', ''):
-        dimension = 'dimensionless'
+            dimension_i = units.get_unit_dimension(attribute_j.unit)
 
     name      = attribute_j.name.strip()
 
-    attr_i = _get_attr_by_name_and_dimension(name, dimension)
+    attr_i = _get_attr_by_name_and_dimension(name, dimension_i.name)
```
Comment (jetuk) [inline hydra_base/lib/template.py]: It is possible that `dimension_i` could be unassigned here.

File: hydra_base/lib/template.py [hunk 5]
```diff
@@ -388,6 +377,8 @@ def get_template_as_xml(template_id,**kwargs):
 
     template_name = etree.SubElement(template_xml, "template_name")
     template_name.text = template_i.name
+    template_name = etree.SubElement(template_xml, "template_description")
```
Comment (jetuk) [inline hydra_base/lib/template.py]: Should be `template_description`?

File: tests/fixtures.py [hunk 6]
```diff
@@ -26,14 +29,14 @@ def testdb_uri(db_backend):
         # This is designed to work on Travis CI
         return 'postgresql://postgres@localhost:5432/hydra_base_test'
     elif db_backend == 'mysql':
-        return 'mysql+mysqldb://root@localhost/hydra_base_test'
+        return 'mysql+mysqldb://root:root@localhost/hydra_base_test'
```
Comment (jetuk) [inline tests/fixtures.py]: This will break travis.

File: tests/fixtures.py [hunk 7]
```diff
@@ -26,14 +29,14 @@ def testdb_uri(db_backend):
...
     else:
         raise ValueError('Database backend "{}" not supported when running the tests.'.format(db_backend))
 
 
 @pytest.fixture(scope='function')
 def engine(testdb_uri):
-    engine = create_engine(testdb_uri)
+    engine = create_engine(testdb_uri, encoding='utf-8')
```
Comment (jetuk) [inline tests/fixtures.py]: I think this is default? But perhaps not in Python 2?

File: tests/fixtures.py [hunk 8]
```diff
@@ -62,6 +65,9 @@ def session(db, engine, request):
     # Patch the global session in hydra_base
     hydra_base.db.DBSession = session
 
+    if six.PY2 and isinstance(session.connection().connection.connection,sqlite3.Connection):
+        session.connection().connection.connection.text_factory = lambda x: unicode(x, 'utf-8', 'ignore')
```
Comment (jetuk) [inline tests/fixtures.py]: `session.connection().connection.connection` ... OK I won't ask why this works!

File: hydra_base/db/model.py (hunk 1 above)
Comment (knoxsp) [reply hydra_base/db/model.py]: Possibly, but this would be the focus of another piece of work

File: tests/fixtures.py (hunk 7 above)
Comment (knoxsp) [reply tests/fixtures.py]: possibly yes but this makes it explicit, so let's leave it.

File: tests/fixtures.py (hunk 8 above)
Comment (knoxsp) [reply tests/fixtures.py]: session.connection()  ==> sqlalchemy connection object
session.connection().connection ==> the connection fairy
session.connection().connection.connection ==> the actual sqlite connection

**Pull request merged**